### ISSUE TITLE
Add methods to request object allowing cookies to be read.

### ISSUE THREAD

**Issue by saeideng** [cakephp/cakephp#9193]
This is a (multiple allowed):
- [ ] bug
- [x] enhancement
- [ ] feature-discussion (RFC)
- CakePHP Version: v3.2.x
### What you did

add new feature to `request::cookie` to return cookie object until we can use cookie component on `$this->request` just like `$this->request->session()`
after that we can use `$this->request->cookie()->read()` , `$this->request->cookie()->write()` ,

**Comment by markstory** [comment]
I don't think it makes sense to write() cookies into a request, but I can see how having a read method would be useful.

**Comment by saeideng** [comment]
You are right
for write method
$this->response->cookie->write();
but probably

**Comment by markstory** [comment]
@saeideng We already have `$response->cookie()` to do writing. I don't know how adding more `->` makes life better for people.

**Comment by saeideng** [comment]
as Cake\Network\Response have not session methods
probably total cookie methods(read/write/delete) can defined within Cake\Network\Request like session method
i think we can not develop $response->cookie()
now,we can not use cookie component within Cells normaly

or use magic getter in php

```
    public function __get($name)
    {
        if($name==='cookie')
           return instance of cookie
        else throw new exception('...')
    }
```
`$this->response->cookie->write();`

but i think must those methods are together in a class (response or request)

**Comment by markstory** [comment]
Why do you need to use CookieComponent in a cell? You have access to the response, allowing cookies to be set/deleted. Its not the best interface, and I agree it could be better, which is what I think this issue is for.

**Comment by saeideng** [comment]
cookies from response is not normal/plane/readable cookie  and that is encrypted

**Comment by markstory** [comment]
Encrypted Cookies will very likely move into a PSR7 Middleware in 3.4, deprecating the CookieComponent. I think this release might also be a good time to consider providing better methods to interact with cookies on the request/response objects.

**Comment by saeideng** [comment]
but probably v 3.4 will release in 4-5 months later

but ok :+1:  
thanks 
you can close this issue

**Comment by markstory** [comment]
@saeideng If ypu don't mind, I would lile to keep it open so work can be tracked in 3.4

**Comment by saeideng** [comment]
Why?

**Comment by markstory** [comment]
@saeideng I closed this because throughout 3.4 several new methods have been added to the response to read/write cookies, and the request already has methods for reading cookies.

I'm not really sold on there needing to be more abstraction on reading request cookies, as they are a simple key/value mapping. We also have separate roadmap items to move encrypted cookies into middleware, so that aspect of this issue will also be addressed.

Is there anything that I've missed/not understood?
